src/java/org/apache/cassandra/cql3/constraints/ConstraintFunction.java

```diff
@@ -34,19 +33,14 @@ public interface ConstraintFunction
     String getName();
 
     /**
-     * Method that provides the execution of the condition. It can either succeed or throw a {@link ConstraintViolationException}.
-     *
-     * @param valueType
-     * @param relationType
-     * @param term
-     * @param columnValue
+     * Method that performs the actual condition test, executed during the write path.
+     * It the test is not successful, it throws a {@link ConstraintViolationException}.
      */
     void evaluate(Class<? extends AbstractType> valueType, Operator relationType, String term, Object columnValue) throws ConstraintViolationException;
```

Comment:
commented else where already, just repeat here. The interface can be updated to 

```suggestion
    void evaluate(AbstractType<?> valueType, ByteBuffer value, Operator relationType, String term) throws ConstraintViolationException;

```